taskcluster/scriptworker_taskgraph/target_tasks.py

```diff
@@ -16,4 +16,18 @@ def filter(task, parameters):
             return False
         if parameters.get("script_name"):
             return task.attributes.get("script-name") == parameters["script_name"]
+        return True
+    return [l for l, t in full_task_graph.tasks.iteritems() if filter(t, parameters)]
+
+
+@_target_task('docker-hub-push')
+def target_tasks_default(full_task_graph, parameters, graph_config):
+    """Filter by `run_on_tasks_for` and `script-name`."""
+
+    def filter(task, parameters):
+        if task.kind != "k8s-image":
+            return False
+        if parameters.get("script_name"):
+            return task.attributes.get("script-name") == parameters["script_name"]
```

Comment:
(optional) Could turn this into a `filter_for_script` function (that returns `True` if the parameter is unset) and then use a style like [this](https://searchfox.org/mozilla-central/source/taskcluster/taskgraph/target_tasks.py#292-295).